File: content/problems/0583-delete-operation-for-two-strings/solutions/lcs-rolling-dp/Solution.java
class Solution {
    public int minDistance(String word1,String word2){
        int[] dp=new int[word2.length()+1];
        for(int i=1;i<=word1.length();i++){
            int diagonal=0;
            for(int j=1;j<=word2.length();j++){
                int above=dp[j];
                if(word1.charAt(i-1)==word2.charAt(j-1))dp[j]=diagonal+1;
                else dp[j]=Math.max(dp[j],dp[j-1]);
                diagonal=above;
            }
        }
        return word1.length()+word2.length()-2*dp[word2.length()];
    }
}

File: content/problems/0583-delete-operation-for-two-strings/solutions/lcs-rolling-dp/solution.cpp
class Solution {
    public: int minDistance(string word1,string word2){
        vector<int> dp(word2.size()+1);
        for(int i=1;i<=(int)word1.size();++i){
            int diagonal=0;
            for(int j=1;j<=(int)word2.size();++j){
                int above=dp[j];
                dp[j]=word1[i-1]==word2[j-1]?diagonal+1:max(dp[j],dp[j-1]);
                diagonal=above;
            }
        }
        return word1.size()+word2.size()-2*dp.back();
    }
};
